sopel/trigger.py

```diff
@@ -38,6 +38,27 @@
 IdentifierFactory = Callable[[str], identifiers.Identifier]
 
 
+COMMANDS_WITH_CONTEXT = [
+    'INVITE',
+    'JOIN',
+    'KICK',
+    'MODE',
+    'NOTICE',
+    'PART',
+    'PRIVMSG',
+    'TOPIC',
+]
```

Comment:
## PERFORMANCE WANKERY WARNING

Is order important here? If not, a `set()` (or `frozenset()`?) might be a small improvement that accelerates membership tests, which are I imagine going to be the typical operations with this.

Because the perf is in the pudding, here's what I think is a representative microbenchmark for `'PART'`:

```17:36 [snoopjedi@denton ~]
$ python3.9 -m timeit -n 1000000 -s "COMMANDS_WITH_CONTEXT = [ 'INVITE', 'JOIN', 'KICK', 'MODE', 'NOTICE', 'PART', 'PRIVMSG', 'TOPIC', ]" "'PART' in COMMANDS_WITH_CONTEXT"
1000000 loops, best of 5: 122 nsec per loop
17:37 [snoopjedi@denton ~]
$ python3.9 -m timeit -n 1000000 -s "COMMANDS_WITH_CONTEXT = { 'INVITE', 'JOIN', 'KICK', 'MODE', 'NOTICE', 'PART', 'PRIVMSG', 'TOPIC', }" "'PART' in COMMANDS_WITH_CONTEXT"
1000000 loops, best of 5: 30.4 nsec per loop
```

Tests for members that aren't present are fairly comparable to this, although early hits in the list do seem to eke out a small victory, so if `PRIVMSG` (or some other member) is the source of _most_ triggers, maybe it makes sense to keep this a list and put that member at the front of it.

Reply:
Want to add tuple to that microbenchmark?

Reply:
Good idea, it's comparable to the `list` spelling (which makes sense, it's basically the same linear scan):

```
17:54 [snoopjedi@denton ~/scratch]
$ python3.9 -m timeit -n 1000000 -s "COMMANDS_WITH_CONTEXT = ( 'INVITE', 'JOIN', 'KICK', 'MODE', 'NOTICE', 'PART', 'PRIVMSG', 'TOPIC', )" "'QUIT' in COMMANDS_WITH_CONTEXT"
1000000 loops, best of 5: 171 nsec per loop
```

Reply:
A set really is the way to go, then. You really can't beat a good old hashtable backend.